src/Mod/Sketcher/Gui/DrawSketchHandlerNote.h

```diff
@@ -0,0 +1,284 @@
+ *   version 2 of the License, or (at your option) any later version.      *
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef SKETCHERGUI_DrawSketchHandlerNote_H
+#define SKETCHERGUI_DrawSketchHandlerNote_H
+
+#include <Gui/BitmapFactory.h>
+#include <Gui/Notifications.h>
+#include <Gui/Command.h>
+#include <Gui/CommandT.h>
+#include <QInputDialog>
+#include <QLineEdit>
+
+#include <Mod/Sketcher/App/SketchObject.h>
+
+#include "GeometryCreationMode.h"
+
+#include "DrawSketchDefaultWidgetController.h"
+#include "DrawSketchControllableHandler.h"
+
+namespace SketcherGui
+{
+
+extern GeometryCreationMode geometryCreationMode;  // defined in CommandCreateGeo.cpp
+
+class DrawSketchHandlerNote;
+
+using DSHNoteController = DrawSketchController<DrawSketchHandlerNote,
+                                               StateMachines::OneSeekEnd,
+                                               /*PAutoConstraintSize =*/1,
+                                               /*OnViewParametersT =*/OnViewParameters<2>>;
+
+using DrawSketchHandlerNoteBase = DrawSketchControllableHandler<DSHNoteController>;
+
+class DrawSketchHandlerNote: public DrawSketchHandlerNoteBase
+{
+    // Allow specialisations of controllers access to private members
+    friend DSHNoteController;
+
+public:
+    DrawSketchHandlerNote() = default;
+    ~DrawSketchHandlerNote() override = default;
+
+private:
+    void updateDataAndDrawToPosition(Base::Vector2d onSketchPos) override
+    {
+        switch (state()) {
+            case SelectMode::SeekFirst: {
+                toolWidgetManager.drawPositionAtCursor(onSketchPos);
+
+                editNote = onSketchPos;
+
+                seekAndRenderAutoConstraint(sugConstraints[0],
+                                            onSketchPos,
+                                            Base::Vector2d(0.f, 0.f));
+            } break;
+            default:
+                break;
+        }
+    }
+
+    void executeCommands() override
+    {
+        try {
+            bool ok;
+            QString text = QInputDialog::getText(nullptr,
```

Comment:
Is that dialog properly parented? I think without a parent it will pop up in the wrong place.